MdeModulePkg/Universal/SetupBrowserDxe/IfrParse.c

```diff
@@ -2166,7 +2268,13 @@ ParseOpCodes (
         // A Question may have more than one Default value which have different default types.
         //
         CurrentDefault = AllocateZeroPool (sizeof (QUESTION_DEFAULT));
-        ASSERT (CurrentDefault != NULL);
+        // MU_CHANGE [BEGIN] - CodeQL change
+        if (CurrentDefault == NULL) {
+          ASSERT (CurrentDefault != NULL);
+          return EFI_OUT_OF_RESOURCES;
+        }
+
+        // MU_CHANGE [END] - CodeQL change

```

Comment:
A lot of these case blocks have a `AllocateCopyPool()` followed by an `ASSERT()`. For example:

```c
          CurrentDefault->Value.Buffer    = AllocateCopyPool (CurrentDefault->Value.BufferLen, &((EFI_IFR_DEFAULT *)OpCodeData)->Value);
          ASSERT (CurrentDefault->Value.Buffer != NULL);
```

If you haven't already, can you check if it is valid for the buffer to be null for a node in the list?